### ISSUE TITLE
Regent: Support for terra bit operations

### ISSUE THREAD

**Issue by LonelyCat124** [StanfordLegion/legion#867]
Legion version: 467ddee
llvm 3.8

Regent doesn't currently appear to support some terra bitwise operations inside `task` functions:
```
import "regent"

local c = regentlib.c

local format = require("std/format")


task main()

  var one : int64 = 1LL
  var two : int64 = 2LL

  var one_xor_two : int64 = one ^ two
  var one_or_two : int64 = one or two
  var one_and_two : int64 = one and two
  var one_lshift_one : int64 = one << 1
  var two_rshift_one : int64 = two >> 1

end

regentlib.start(main)
```

I commented out the bitwise statements one by one for the following outputs:
All:
```
bash-4.2$ legion/language/regent.py test_bitwise.rg
test_bitwise.rg:13: unexpected token in expression near '^'
```

Commented out XOR:
```
bash-4.2$ legion/language/regent.py test_bitwise.rg
test_bitwise.rg:16: unexpected token in expression near '<<'
```

Commented out xor, and lshift
```
bash-4.2$ legion/language/regent.py test_bitwise.rg
test_bitwise.rg:17: unexpected token in expression near '>>'
```

Commented out xor, lshift and rshift
Code compiles - adding print statements for remaining variables:
```
bash-4.2$ legion/language/regent.py test_bitwise.rg
one or two = 3
one and two = 0
```

It looks like bitwise and and bitwise or are working correctly, but I don't think regent is parsing xor or shift operations in task code (wheras in terra blocks they work fine for me). I'm not sure if they collide with symbols used for other things in Regent - (I didn't find anything in the docs).

**Comment by elliottslaughter** [comment]
Yes, we should add these.

**Comment by rupanshusoi** [comment]
@elliottslaughter I've added support for these bit operations in a PR, please check.

**Comment by elliottslaughter** [comment]
This has been merged with #870. Thanks @rupanshusoi !
